Importing libraries

In [18]:
import os
import glob
import pandas as pd 
import csv
import collections
import pprint

We display the preprocessed .csv files

In [4]:
data_directory = '../Ivan Compagnucci/EDA-EnterpriseDataAnalytics-project/Data/Merged data/'
raw_files = os.listdir(data_directory)

raw_files

['environmental.csv',
 'luxmeter2018.csv',
 'luxmeter2019.csv',
 'powerelectricity.csv',
 'presence2018.csv',
 'presence2019.csv',
 'reedsensor.csv']

<h3>Now we will verify the integrity of the data of each .csv file</h3> 
<h4 style="color:red"> Environmental.csv </h4>

In [15]:
df = pd.read_csv("../Ivan Compagnucci/EDA-EnterpriseDataAnalytics-project/Data/Merged data/environmental.csv")
df.info()
singleDataNull = sum(df.isnull().values.ravel())
singleRowContainingNull = sum([True for idx,row in df.iterrows() if any(row.isnull())])

print('We have',singleDataNull,'null values in',singleRowContainingNull,'rows')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48173 entries, 0 to 48172
Data columns (total 17 columns):
idsensor                     48173 non-null object
sensortype                   48173 non-null object
srtDate                      48173 non-null object
absolute_humidity            48173 non-null float64
back_front                   48173 non-null float64
battery_level                48173 non-null float64
device_temperature           48044 non-null float64
dew_point                    48044 non-null float64
environmental_temperature    48044 non-null float64
movement_level               48044 non-null float64
orientation                  48044 non-null float64
position                     48044 non-null float64
posture_tilt                 48044 non-null float64
pressure                     48044 non-null float64
regular                      48044 non-null float64
relative_humidity            48044 non-null float64
sequence_number              48044 non-null float64
dtypes: fl

Due to the lack of device data in 129 records, I decided to exclude these rows.
Since in total only 129 lines out of 48173 contain a null value, we can proceed with the elimination of the latter
then saving the .csv file

In [22]:
df.dropna(inplace=True)
df.to_csv('../Ivan Compagnucci/EDA-EnterpriseDataAnalytics-project/Data/Cleaned data/environmental.csv')